generic-file-system/api/src/main/java/org/pentaho/platform/api/genericfile/GetTreeOptions.java

```diff
@@ -96,7 +97,7 @@ public GetTreeOptions( @NonNull GetTreeOptions other ) {
 
     this.basePath = other.basePath;
     this.maxDepth = other.maxDepth;
-    this.expandedPath = other.expandedPath;
+    this.expandedPaths = other.expandedPaths != null ? List.copyOf( other.expandedPaths ) : null;
```

Comment:
Why is it necessary to copy the array in this case? Since the type is an object, would it be better to perform a deep copy instead?

Reply:
IIRC, this was a matter of defensive programming. We want to become independent of the given `GetTreeOptions` instance, and that implies not assuming its list won't be modified.

Generic file paths are immutable, so no need for a deep copy.